query/src/sql/parser/transformer/transform_sqlparser.rs

```diff
@@ -79,12 +79,190 @@ impl TransformerSqlparser {
         }
     }
 
-    fn transform_ddl(&self, _: &SqlparserStatement) -> Result<Statement> {
-        // TODO: support transform DDL
-        Err(ErrorCode::SyntaxException(format!(
-            "Unsupported SQL statement: {}",
-            self.orig_stmt
-        )))
+    fn get_database_and_table_from_idents(
+        &self,
+        idents: &[Ident],
+    ) -> (Option<Identifier>, Option<Identifier>) {
+        if idents.len() == 1 {
+            (None, Some(Identifier::from(&idents[0])))
+        } else if idents.len() == 2 {
+            (
+                Some(Identifier::from(&idents[0])),
+                Some(Identifier::from(&idents[0])),
+            )
+        } else {
+            (None, None)
+        }
+    }
+
+    fn transform_ddl(&self, orig_ast: &SqlparserStatement) -> Result<Statement> {
+        match orig_ast {
+            SqlparserStatement::Truncate { table_name, .. } => {
+                let (db, tpl) = self.get_database_and_table_from_idents(&table_name.0);
+                if tpl.is_none() {
+                    return Err(ErrorCode::SyntaxException(format!(
+                        "Unsupported SQL statement: {}",
+                        self.orig_stmt
+                    )));
+                }
+                Ok(Statement::TruncateTable {
+                    database: db,
+                    table: tpl.unwrap(),
+                })
+            }
+            SqlparserStatement::CreateTable {
+                if_not_exists,
+                name,
+                columns,
+                ..
+            } => {
+                let (db, tpl) = self.get_database_and_table_from_idents(&name.0);
+                if tpl.is_none() {
+                    return Err(ErrorCode::SyntaxException(format!(
+                        "Unsupported SQL statement: {}",
+                        self.orig_stmt
+                    )));
+                }
```

Comment:
It's better to unwrap `tpl` here like:
```suggestion
                let tpl = tpl.ok_or(Err(ErrorCode::SyntaxException(format!(
                        "Unsupported SQL statement: {}",
                        self.orig_stmt
                    ))))?;
```

So we don't need to handle the `None` case latter.